_overviews/scala3-book/scala-for-python-devs.md

```diff
@@ -0,0 +1,1041 @@
+      <td valign="top"><code>val t = (11, 11.0, "Eleven")</code></td>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Access elements</td>
+    </tr>
+    <tr>
+      <td valign="top"><code>t[0]&nbsp;&nbsp; # 11
+        <br>t[1]&nbsp;&nbsp; # 11.0</code>
+      </td>
+      <td valign="top"><code>t(0)&nbsp;&nbsp; // 11
+        <br>t(1)&nbsp;&nbsp; // 11.0</code>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+
+
+## Methods on collections classes
+
+Python and Scala have several of the same common functional methods available to them:
+
+- `map`
+- `filter`
+- `reduce`  
+
+If you’re used to using these methods with lambda expressions in Python, you’ll see that Scala has a similar approach with methods on its collections classes. To demonstrate this functionality, here are two sample lists:  
+
+```scala
+list = (1,2,3)           // python
+val list = List(1,2,3)   // scala
+```
+
+Those lists are used in the following table, that shows how to apply mapping and filtering algorithms to it:
+
+<table cellspacing="1" cellpadding="2" border="1">
+  <tbody>
+    <tr>
+      <th valign="top">Python</th>
+      <th valign="top">Scala</th>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Mapping with a comprehension</td>
+    </tr>
+    <tr>
+      <td valign="top"><code>x = [i*10 for i in list]</code></td>
+      <td valign="top"><code>val x = for i &lt;- list yield i * 10</code></td>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Filtering with a comprehension</td>
+    </tr>
+    <tr>
+      <td valign="top"><code>evens = [i for i in list if i % 2 == 0]</code></td>
+      <td valign="top"><code>val evens = list.filter(_ % 2 == 0)</code></td>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Mapping and filtering with a comprehension</td>
+    </tr>
+    <tr>
+      <td valign="top"><code>x = [i * 10 for i in list if i % 2 == 0]</code></td>
+      <td valign="top"><code>val x = xs.filter(_ % 2 == 0)
+        <br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.map(_ * 10)</code>
+      </td>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Mapping with <code>map</code></td>
+    </tr>
+    <tr>
+      <td valign="top"><code>def times_10(n): return n * 10
+        <br>x = map(times_10, list)</code>
+      </td>
+      <td valign="top"><code>val x = list.map(_ * 10)</code></td>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Filtering with <code>filter</code></td>
+    </tr>
+    <tr>
+      <td valign="top"><code>f = lambda x: x if x &gt; 1 else 1
+        <br>x = filter(f, list)</code>
+      </td>
+      <td valign="top"><code>val x = list.filter(_ &gt; 1)</code></td>
+    </tr>
+  </tbody>
+</table>
+
+Scala collections classes have over 100 functional methods to simplify your code. In addition to `map`, `filter`, and `reduce`, other commonly-used methods are listed below.
+
+Filtering methods:
+
+- `diff`
```

Comment:
We should probably add a short note to each if we are going to name them. Or link to the documentation.